rewrite-xml/src/main/java/org/openrewrite/xml/trait/SpringTypeReference.java

```diff
@@ -52,6 +50,27 @@ public String getName() {
         throw new IllegalArgumentException("getTree() must be an Xml.Attribute or Xml.Tag: " + getTree().getClass());
     }
 
+    @Override
+    public boolean supportsRename() {
+        return true;
+    }
+
+    @Override
+    public TreeVisitor<Tree, ExecutionContext> renameTo(String name) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext executionContext) {
+                if (tree instanceof Xml.Attribute) {
+                    return ((Xml.Attribute) tree).withValue(((Xml.Attribute) tree).getValue().withValue(name));
+                }
+                if (tree instanceof Xml.Tag) {
+                    return ((Xml.Tag) tree).withValue(name);
+                }
+                return super.visit(tree, executionContext);
+            }
+        };
+    }
+
     static class Matcher extends SimpleTraitMatcher<SpringTypeReference> {
         private final Pattern typeReference = Pattern.compile("(?:[a-zA-Z_][a-zA-Z0-9_]*\\.)+[A-Z*][a-zA-Z0-9_]*(?:<[a-zA-Z0-9_,?<> ]*>)?");
```

Comment:
The regex isnot fully covering just yet: 
![image](https://github.com/user-attachments/assets/424dce1b-1643-44e4-8666-469a4500c242)

Reply:
Yes, we might indeed have to cover generics, too. Otherwise this would also work: `"\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*(?:\\.\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)*"`.

Reply:
I had never seen `\p{javaJavaIdentifierStart}` before but it seems that matches what we want to match here much better!